dtfs-central-api/src/v1/routes/utilisation-reports-routes.js

```diff
@@ -399,11 +399,55 @@ utilisationReportsRouter.route('/:reportId/fee-records-to-key').get(validation.s
  *         description: Not Found
  *       500:
  *         description: Internal Server Error
+ *   patch:
+ *     summary: Edit the payment
+ *     tags: [Utilisation Report]
+ *     description: Edits the payment with the supplied id
+ *     parameters:
+ *       - in: path
+ *         name: reportId
+ *         schema:
+ *           type: string
+ *         required: true
+ *         description: the id for the report
+ *       - in: path
+ *         name: paymentId
+ *         schema:
+ *           type: string
+ *         required: true
+ *         description: the id for the payment
+ *     requestBody:
+ *       required: true
+ *       content:
+ *         application/json:
+ *           schema:
+ *             type: object
+ *             properties:
+ *               paymentAmount:
+ *                 type: number
+ *               datePaymentReceived:
+ *                 type: string
+ *                 format: date
+ *               paymentReference:
+ *                 type: string
+ *                 required: false
```

Comment:
This is interesting, because when not provided it updates the field to null from a value if one already exists, which might not be the default expectation for a PATCH when a field is not required, what do you think? But then given how JSON works if undefined it just won't be in the body right so I do think this type is correct? Can we add a description or something to make it clear that not providing one unsets it?

Reply:
After a small but of thinking, I reckon we should make this field required and have it be either `null` or `string`. If it is `null`, we set the field to `undefined` (which can be handled by the zod schema). Doing that probably makes the intention clearer

Reply:
As a side note for future readers, there is a reason why we used `undefined` instead of `null` for the typeorm entities - SQL Server processes the javascript `null` value as an `object`, which can't be inserted into SQL Server tables, therefore throwing an error. If you want to set a column to `NULL`, you need to use `undefined` in SQL Server